pkg/service/credential/service.go

```diff
@@ -90,10 +90,30 @@ func NewCredentialService(config config.CredentialServiceConfig, s storage.Servi
 }
 
 func (s Service) CreateCredential(ctx context.Context, request CreateCredentialRequest) (*CreateCredentialResponse, error) {
+	returnFunc := s.CreateCredentialFunc(request)
+	returnValue, err := s.storage.db.Execute(ctx, returnFunc)
 
+	if err != nil {
+		return nil, errors.Wrap(err, "execute")
+	}
+
+	credResponse, ok := returnValue.(*CreateCredentialResponse)
+	if !ok {
+		return nil, errors.New("Problem with casting to CreateCredentialResponse")
+	}
+
+	return credResponse, nil
+}
+
+func (s Service) CreateCredentialFunc(request CreateCredentialRequest) storage.BusinessLogicFunc {
+	return func(ctx *context.Context) (any, error) {
+		return s.CreateCredentialBusinessLogic(ctx, request)
+	}
+}
+
+func (s Service) CreateCredentialBusinessLogic(ctx *context.Context, request CreateCredentialRequest) (*CreateCredentialResponse, error) {
```

Comment:
why does this accept a context pointer but other methods do not? let's be consistent

Reply:
to be clearer I don't think we should have ptrs

Reply:
yup I was trying to overload this CTX object instead of making my own accumulator object, I have removed and no more ctx pointers